UriMatchService/UriTemplateMatcher.cs

```diff
@@ -69,13 +69,20 @@ public TemplateMatch Match(Uri uri)
             var templateMatches = new Dictionary<KeyValuePair<string, string>, int>(); // {{ templateKey, foundParameterCount }}
             foreach (var template in _templates)
             {
-                var parameters = GetParameters(absolutePath, template.Value);
-                if (parameters != null)
+                try
                 {
-                    if (parameters.Count == 0)
-                        return new TemplateMatch() { Key = template.Key, Template = template.Value }; // exact match, no ids
-                    else 
-                        templateMatches.Add(template, parameters.Count);
+                    var parameters = GetParameters(absolutePath, template.Value);
+                    if (parameters != null)
+                    {
+                        if (parameters.Count == 0)
+                            return new TemplateMatch() { Key = template.Key, Template = template.Value }; // exact match, no ids
+                        else
+                            templateMatches.Add(template, parameters.Count);
```

Comment:
```suggestion
                        else if(!templateMatches.ContainsKey(template))
                            templateMatches.Add(template, parameters.Count);
```
If we are simply avoiding the throwing of the `ArgumentExcpetion` we could check if the key exists to avoid the exception handling overhead here.

Reply:
@millicentachieng what example scenarios throw exceptions?

Reply:
@irvinesunday Like this one `/employeeexperience/learningproviders/{id}/learningcourseactivities(extern`. There's no closing bracket.

Reply:
Then this makes sense. 
Perhaps we could incorporate the other suggested additions to make this more robust?

Reply:
Just to confirm, do we still need the try catch block after the change? @millicentachieng